go/stellar/stellarsvc/frontend.go

```diff
@@ -239,16 +243,209 @@ func (s *Server) LinkNewWalletAccountLocal(ctx context.Context, arg stellar1.Lin
+	default:
+		return nil, fmt.Errorf("unrecognized payment type: %s", typ)
+	}
+}
+
+func (s *Server) transformPaymentStellar(ctx context.Context, p stellar1.PaymentSummaryStellar) (*stellar1.PaymentLocal, error) {
+	loc, err := newPaymentLocal(p.TxID, p.Ctime, p.Amount)
+	if err != nil {
+		return nil, err
+	}
+
+	loc.Source = p.From.String()
+	loc.SourceType = ParticipantTypeStellar
+	loc.Target = p.To.String()
+	loc.TargetType = ParticipantTypeStellar
+
+	loc.Status, loc.StatusDetail = stellar1.TransactionStatus_SUCCESS.Details("")
+
+	return loc, nil
+}
+
+func (s *Server) transformPaymentDirect(ctx context.Context, p stellar1.PaymentSummaryDirect) (*stellar1.PaymentLocal, error) {
+	loc, err := newPaymentLocal(p.TxID, p.Ctime, p.Amount)
+	if err != nil {
+		return nil, err
+	}
+
+	loc.Worth, loc.WorthCurrency, err = s.formatWorth(ctx, p.DisplayAmount, p.DisplayCurrency)
+	if err != nil {
+		return nil, err
+	}
+
+	loc.Source, loc.SourceType = s.lookupUsernameFallback(ctx, p.From.Uid, p.FromStellar)
+
+	if p.To != nil {
+		loc.Target, loc.TargetType = s.lookupUsernameFallback(ctx, p.To.Uid, p.ToStellar)
+	}
+
+	loc.Status, loc.StatusDetail = p.TxStatus.Details(p.TxErrMsg)
+
+	loc.Note, loc.NoteErr = s.decryptNote(ctx, p.TxID, p.NoteB64)
+
+	return loc, nil
+}
+
+func (s *Server) transformPaymentRelay(ctx context.Context, p stellar1.PaymentSummaryRelay) (*stellar1.PaymentLocal, error) {
+	loc, err := newPaymentLocal(p.TxID, p.Ctime, p.Amount)
+	if err != nil {
+		return nil, err
+	}
+
+	loc.Worth, loc.WorthCurrency, err = s.formatWorth(ctx, p.DisplayAmount, p.DisplayCurrency)
+	if err != nil {
+		return nil, err
+	}
+
+	loc.Source, loc.SourceType = s.lookupUsernameFallback(ctx, p.From.Uid, p.FromStellar)
+
+	if p.To == nil {
+		return nil, errors.New("no recipient")
+	}
+	name, err := s.lookupUsername(ctx, p.To.Uid)
+	if err != nil {
+		s.G().Log.CDebugf(ctx, "recipient lookup failed: %s", err)
+		return nil, errors.New("recipient lookup failed")
+	}
+	loc.Target = name
+	loc.TargetType = ParticipantTypeKeybase
+
+	if p.TxStatus != stellar1.TransactionStatus_SUCCESS {
+		// If the funding tx is not complete
+		loc.Status, loc.StatusDetail = p.TxStatus.Details(p.TxErrMsg)
+	} else {
+		loc.Status = "claimable"
+
+		loc.StatusDetail = "Waiting for the recipient to open the app to claim, or the sender to yank."
+	}
+	if p.Claim != nil {
+		if p.Claim.TxStatus == stellar1.TransactionStatus_SUCCESS {
+			// If the claim succeeded, the relay payment is done.
+			loc.Status = "completed"
+			name, err := s.lookupUsername(ctx, p.Claim.To.Uid)
+			if err == nil {
+				loc.Target = name
+				loc.TargetType = ParticipantTypeKeybase
+			} else {
+				loc.Target = p.Claim.ToStellar.String()
+				loc.TargetType = ParticipantTypeStellar
+			}
+		} else {
+			claimantUsername, err := s.lookupUsername(ctx, p.Claim.To.Uid)
+			if err != nil {
+				return nil, err
+			}
+			loc.Status, loc.StatusDetail = p.Claim.TxStatus.Details(p.Claim.TxErrMsg)
```

Comment:
Setting loc.Status here may not be the right move. I see that `localizePayment` does it too. It's just that in this case the relay is still claimable, and there should be a cancel button for the user. So if status drives that this would result in stuck money.